bmwqemu.pm

```diff
@@ -316,6 +320,39 @@ sub TIEHASH ($class, %args) {
 sub STORE ($self, $key, $val) {
     croak("Settings key '$key' is invalid (check your test settings)") unless $key =~ m/^(?:[A-Z0-9_]+)\z/;
     $self->{data}->{$key} = $val;
+    my @stack;
+    for (0 .. 9) {
+        my @caller = caller($_);
+        last unless @caller;
+        my ($package, $file, $line, $sub) = @caller;
+        if ($caller[3] eq 'bmwqemu::load_vars') {
+            $self->{trace}->{load_vars}->{$key} = $val;
+            return;
+        }
+        next if ($package eq 'testapi');
+        last if ($package eq 'basetest' and $sub eq '(eval)');
+        last if $package eq 'main';
+        push @stack, [$package, $line, $sub];
+    }
+    $self->{trace}->{override}->{$key} = {
+        callstack => \@stack,
+        value => $val,
+    };
+}
+
+sub debug_vars ($self) {
```

Comment:
should exist a condition to return based on a env variable? or do you want this running unconditional? I am thinking something like `return unless ENV_TRACE_ENABLED=1` or something